lib/redis_memo/memoizable/dependency.rb

```diff
@@ -26,14 +26,18 @@ def depends_on(dependency, **conditions)
         instance_exec(&memo.depends_on)
       end
     when ActiveRecord::Relation
-      extracted = extract_dependencies_for_relation(dependency)
+      extracted = self.class.extract_from_relation(dependency)
       nodes.merge!(extracted.nodes)
-    when UsingActiveRecord
-      [
-        dependency.redis_memo_class_memoizable,
-        RedisMemo::MemoizeQuery.create_memo(dependency, **conditions),
-      ].each do |memo|
-        nodes[memo.cache_key] = memo
+    when RedisMemo::MemoizeQuery::CachedSelect::BindParams
+      # A private API
+      dependency.params.each do |model, attrs_set|
+        attrs_set.each do |attrs|
+          memo = model.redis_memo_class_memoizable
+          nodes[memo.cache_key] = memo
```

Comment:
I think we can move this out of the inner loop?

Reply:
Ah, that is true!